### PR TITLE
Fix mac m1 chip build, get architecture during make mac-app. Fixes #2160

### LINKED ISSUE #2160: Bug: make mac-app not compatible with m1 chip
## Steps to reproduce the behavior

Run the the command `make mac-app` using a MacBook Pro - Apple M1 Chip . 

Got errors: 

```
% make mac-app
mkdir -p bin/mac
cd server; env GOOS=darwin GOARCH=amd64 go build -ldflags '-X "github.com/mattermost/focalboard/server/model.BuildNumber=dev" -X "github.com/mattermost/focalboard/server/model.BuildDate=n/a" -X "github.com/mattermost/focalboard/server/model.BuildHash=92eb78357fbed987ef412d49667f5ea8f2ae7c47" -X "github.com/mattermost/focalboard/server/model.Edition=mac"' -o ../bin/mac/focalboard-server ./main
package github.com/mattermost/focalboard/server/main: build constraints exclude all Go files in /Users/joaoaldeano/personal/focalboard/server/main
make: *** [server-mac] Error 1
```

## Expected behavior

Looking to ReadMe this command should create an application: `Focalboard.app`.

## Edition and Platform

 - Edition: MacBook Pro (13-inch, M1, 2020) 16 GB
 - Chip: Apple M1 

## Additional context

Looking to the error it seems that Makefile line 36 defines:
`cd server; env GOOS=darwin GOARCH=amd64 go build -ldflags '$(LDFLAGS)' -o ../bin/mac/focalboard-server ./main`

But the systems arch on m1 chip is arm64.

### PR DESCRIPTION
**Pull request by jpaldeano**
#### Ticket Link

Hi again! I had some issues running the `make mac-app` on my mac with the m1 chip - it is building the app here now. Would be nice to have some feedback from people with a amd64 GOARCH to make sure I didn't break anything.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -16,6 +16,13 @@ LDFLAGS += -X "github.com/mattermost/focalboard/server/model.BuildNumber=$(BUILD
 LDFLAGS += -X "github.com/mattermost/focalboard/server/model.BuildDate=$(BUILD_DATE)"
 LDFLAGS += -X "github.com/mattermost/focalboard/server/model.BuildHash=$(BUILD_HASH)"
 
+# MAC cpu architecture
+ifeq ($(shell uname -m),arm64)
+	MAC_GO_ARCH := arm64
+else
+	MAC_GO_ARCH := amd64
```
Comment (jpaldeano) [inline Makefile:23]: ci server reporting `$(shell uname -m)` -> x86_64 which doesn't seem to be valid.

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:    I don't have an M1 to test.

**Review by jespino — APPROVED** [review]
Looks good to me. Thanks @jpaldeano! :confetti_ball:

**Comment by jespino** [comment]
Tested on M1 mac and working as expected :)

**Pull request merged**